arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,253 @@
+    func showEditableAttributes(selectedFeature: AGSFeature) {
+        // Create objects that observe changes to the text of the secure portal URL text field.
+        var streetTextDidChangeObserver: NSObjectProtocol!
+        var addressTextDidChangeObserver: NSObjectProtocol!
+        // Create an alert controller and customize the title and message.
+        let alert = UIAlertController(title: "Edit Feature Attributes", message: "Edit the 'AD_ADDRESS' and 'ST_STR_NAM' attributes.", preferredStyle: .alert)
+        // Add a "Done" option to complete the editing process and close the alert.
+        let doneAction = UIAlertAction(title: "Done", style: .default) { _ in
+            let addressTextField = alert.textFields?[0]
+            let streetTextField = alert.textFields?[1]
+            // Set the new values for the attributes and update the selected feature.
+            selectedFeature.attributes["AD_ADDRESS"] = Int((addressTextField?.text)!)
+            selectedFeature.attributes["ST_STR_NAM"] = streetTextField?.text
+            selectedFeature.featureTable?.update(selectedFeature)
+
+            if let streetObserver = streetTextDidChangeObserver, let addressObserver = addressTextDidChangeObserver {
+                NotificationCenter.default.removeObserver(streetObserver)
+                NotificationCenter.default.removeObserver(addressObserver)
+                streetTextDidChangeObserver = nil
+                addressTextDidChangeObserver = nil
+            }
+        }
+        alert.addAction(doneAction)
+        // Add a text field to prompt the user to input the street address.
+        alert.addTextField { (textField) in
+            // Prompt a number pad for the address.
+            textField.keyboardType = .numberPad
+            // Populate the text fields with the current value.
+            textField.text = (selectedFeature.attributes["AD_ADDRESS"] as! NSNumber).stringValue
+            // Add an observer to ensure the user does not input an empty string.
+            streetTextDidChangeObserver = NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: textField, queue: .main, using: {_ in
+                // Enable the done button if both textfields are not empty.
+                doneAction.isEnabled = (alert.textFields?.allSatisfy { $0.text?.isEmpty == false })!
+            })
+        }
+        // Add a text field to prompt the user to input the street name.
+        alert.addTextField { (textField) in
+            // Prompt a keyboard to enter the street name.
+            textField.keyboardType = .default
+            textField.text = selectedFeature.attributes["ST_STR_NAM"] as? String
+            // Add an observer to ensure the user does not input an empty string.
+            addressTextDidChangeObserver = NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: textField, queue: OperationQueue.main, using: {_ in
```

Comment:
There's actually no need for a second observer. Just pass `nil` for the object and the same closure will be evaluated when the text of either field changes.